src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,97 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        checkThreshold();
+        int index = indexOfBucket(key);
+        Node<K, V> node = table[index];
+        if (table[index] == null) {
+            table[index] = new Node<>(hash(key), key, value, null);
+            size++;
+            return;
+        }
+        while (node != null) {
+            if (Objects.equals(key, node.key)) {
+                node.value = value;
+                return;
+            }
+            if (node.next == null) {
+                node.next = new Node<>(hash(key), key, value, null);
+                size++;
+                return;
+            }
+            node = node.next;
+        }
 
     }
 
     @Override
     public V getValue(K key) {
+        int index = indexOfBucket(key);
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (Objects.equals(key, node.key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void checkThreshold() {
+        int threshold = (int) (table.length * DEFAULT_LOAD_FACTOR);
+        if (size > threshold) {
+            resize();
+        }
+    }
+
+    private int hash(Object key) {
+        int h;
+        return (key == null) ? 0 : (h = key.hashCode()) ^ (h >>> 16);
+    }
+
+    private int indexOfBucket(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode() % table.length);
+    }
+
+    private void resize() {
+        Node<K, V>[] oldTable = table;
+        table = new Node[oldTable.length * 2];
+        size = 0;
+        for (Node<K, V> node : oldTable) {
+            while (node != null) {
+                put(node.key, node.value);
+                node = node.next;
+            }
+        }
+    }
+
+    private static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(int hash, K key, V value, Node<K, V> next) {
```

Comment:
```suggestion
        Node(K key, V value, Node<K, V> next) {
```
you don't initialise hash field in this constructor, moreover we don't need hash code field in Node class